lib/service_manager.dart

```diff
@@ -81,6 +92,29 @@ class ServiceConnectionManager {
     }
   }
 
+  StreamSubscription<bool> hasRegisteredService(
+    String name,
+    void onData(bool value),
+  ) {
+    if (registeredMethodsForService.containsKey(name) && onData != null) {
+      onData(true);
+    }
+    final StreamController<bool> streamController =
+        _getServiceRegistrationController(name);
+    return streamController.stream.listen(onData);
+  }
+
+  StreamController<bool> _getServiceRegistrationController(String name) {
```

Comment:
nit: maybe call this 
_getServiceRegistrationStream
as the helper is called
_getStream

Reply:
Perhaps I should rename _getStream to _getStreamController because it returns a StreamController and not a stream.